src/qir2qasm_trans/qir_trans/translator.py

```diff
@@ -0,0 +1,343 @@
+                    )
+                )
+
+        return statements
+
+    def build_func_declarations(self) -> List[ast.Statement]:
+        module = self.module
+
+        statements_declaration = []
+        statements_definition = []
+
+        for name, struct_info in self.profile.structures.items():
+            self.symbols.register_structure(name, struct_info)  # Register Structure in SymbolTable
+
+        for name, inst_info in self.profile.classical_instruction.items():
+            self.symbols.instructions[name] = inst_info  # Register Instruction in SymbolTable
+
+        for func in module.functions:
+            if func.is_declaration:
+                func_info = self.profile.standard_functions.get(func.name)
+
+                if func_info:
+                    # If func is defined in QIR Profile
+                    self.symbols.functions[func.name] = (
+                        func_info  # Register Function in SymbolTable
+                    )
+                    def_statement = func_info.def_statement
+                    if def_statement:
+                        statements_definition.append(def_statement)
+                else:
+                    # If func is not defined in QIR Profile
+                    func_name = func.name
+                    func_type = func.type.element_type
+
+                    # Build the "defcal" QASM ast node
+                    arguments = []
+                    qubits = []
+                    num_qubits = 0
+
+                    arg_types = [arg_type for arg_type in func_type.elements]
+
+                    # Return Type
+                    ret_type = arg_types[0]
+                    _, ret_type_ast = self.symbols.type_qir2qasm(ret_type)
+                    if isinstance(ret_type_ast, str):
+                        ret_type_ast = self.symbols.structures[ret_type_ast].type_ast
+                    if ret_type_ast == "Qubit":
+                        # Qubit* func(...) => void func (..) Qubit
+                        qubits.append(ast.Identifier(name="q_ret"))
+                        ret_type_ast = None
+
+                    # Argument Type
+                    for op_type in arg_types[1:]:
+                        op_type_str, op_type_ast = self.symbols.type_qir2qasm(op_type)
+                        if isinstance(op_type_ast, str):
+                            op_type_ast = self.symbols.structures[op_type_ast].type_ast
+                        if op_type_ast == "Qubit":
+                            num_qubits += 1
+                        else:
+                            arguments.append(op_type_ast)
+
+                        if (op_type_str == "pointer") and isinstance(
+                            op_type_ast, ast.ClassicalType
+                        ):
+                            if ret_type_ast is None:
+                                # void func(typeA*, ...) => typeA func(type_A, ...)
+                                ret_type_ast = op_type_ast
```

Comment:
I don;t understand this. What is the use case? 
If there is no return, the arg become the return? It does not make much sense to me.